exporter/chronicleexporter/config.go

```diff
@@ -0,0 +1,78 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chronicleexporter
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+// Alternative regional endpoints for Chronicle.
+// https://cloud.google.com/chronicle/docs/reference/search-api#regional_endpoints
+var regions = map[string]string{
+	"Europe Multi-Region":        "https://europe-backstory.googleapis.com",
+	"Frankfurt":                  "https://europe-west3-backstory.googleapis.com",
+	"London":                     "http://europe-west2-backstory.googleapis.com",
+	"Singapore":                  "https://asia-southeast1-backstory.googleapis.com",
+	"Sydney":                     "https://australia-southeast1-backstory.googleapis.com",
+	"Tel Aviv":                   "https://me-west1-backstory.googleapis.com",
+	"United States Multi-Region": "https://united-states-backstory.googleapis.com",
+	"Zurich":                     "https://europe-west6-backstory.googleapis.com",
+}
+
+// Config defines configuration for the Chronicle exporter.
+type Config struct {
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+
+	// Endpoint is the URL where Chronicle data will be sent.
+	Region string `mapstructure:"region"`
+
+	// CredsFilePath is the file path to the Google credentials JSON file.
+	CredsFilePath string `mapstructure:"creds_file_path"`
+
+	// Creds are the Google credentials JSON file.
+	Creds string `mapstructure:"creds"`
+
+	// LogType is the type of log that will be sent to Chronicle.
+	LogType string `mapstructure:"log_type"`
+
+	// RawLogField is the field name that will be used to send raw logs to Chronicle.
+	RawLogField string `mapstructure:"raw_log_field"`
```

Comment:
Thoughts on making this an OTTL expression (https://github.com/observIQ/bindplane-agent/blob/f484d280bd5e105de4bde4dbc7e6a284c3b5de32/expr/ottl_expression.go#L71)? Then parsing and evaluating the expression would be easy.
We should also validate this field in Validation.

Reply:
Updated for most of it but is there anything I'm missing in the switch case for the types?

Reply:
Discussed this offline, the current cases for the switch seems fine, we can add more in later if needed.